andela/ah-backend-poseidon
# 162163014 Estimate an article's reading time

#### What does this PR do?
- Adds a read time to an article.

#### Description of Task to be completed?
- A user is able to view the reading time of an article.

#### How should this be manually tested?
First clone this repository
```
$ git clone https://github.com/andela/ah-backend-poseidon
$ cd ah-backend-poseidon
```
Create virtual environment and install it
```
$ python3 -m venv venv
$ source/venv/bin/activate
```
Install all the necessary dependencies
```
pip install -r requirements.txt
```
Create a Postgres database called ahdb and make migrations by running:
```
 $ python manage.py makemigrations
 $ python manage.py migrate
 $ python manage.py runserver
```
Using swagger docs on (http://127.0.0.1:8000/docs/), try out these endpoints
Retrieve all articles
```GET /api/articles```

Retrieve a single article
```GET /api/articles/{slug}```

Example response:
```{
  "articles": [
    {
      "slug": "the-origins-of-earths-water-questioned",
      "title": "The origins of Earth’s water questioned",
      "description": "Earth’s water may have originated from both asteroids and gas left over from the formation of the Sun according to new research. The discovery has profound implications for both the origins of life on Earth and the possibility of life elsewhere in the Universe.",
      "body": "The study, published in Journal of Geophysical Research: Planets, casts doubt on previously held belief of the source of Earth’s water by suggesting that the hydrogen which comprises the majority of the water molecule may have come from the solar nebula, clouds of gas and dust left over from our star’s formation. Previous models have connected the water found on Earth to that found on asteroids implying that the majority of water delivered to the planet’s surface was a result of impacts from these objects. The ration of deuterium, a heavier isotope of hydrogen, in seawater is similar to the ratio found on asteroids. Scientists have traditionally used this method of tracing hydrogen to discover water’s origins simply because it is the most abundant element in the universe. To conduct their investigation the team led by Peter Buseck, Regents’ Professor in the School of Earth and Space Exploration and School of Molecular Sciences at Arizona State University, took a different approach. Steven Desch, a professor of astrophysics in the School of Earth and Space Exploration at Arizona State University in Tempe, Arizona and co-author of the new study, explains: “When people measure the [deuterium-to-hydrogen] ratio in ocean water and they see that it is pretty close to what we see in asteroids, it was always easy to believe it all came from asteroids.” The authors of the paper went further, suggesting that hydrogen in Earth’s oceans does not represent hydrogen throughout the entire planet. Samples of hydrogen from deep inside the Earth, close to the boundary between the core and mantle, have notably less deuterium, indicating this hydrogen may not have come from asteroids. Noble gases helium and neon, with isotopic signatures inherited from the solar nebula, have also been found in the Earth’s mantle. According to the new model suggested in the paper, several billion years ago, large waterlogged asteroids began developing into planets while the solar nebula still swirled around the Sun. These asteroids, known as planetary embryos, collided and grew rapidly. Eventually, a collision introduced enough energy to melt the surface of the largest embryo into an ocean of magma. This largest embryo would eventually become Earth. Gases from the solar nebula, including hydrogen and noble gases, were drawn in by the large, magma-covered embryo to form an early atmosphere. Nebular hydrogen, which contains less deuterium and is lighter than asteroidal hydrogen, dissolved into the molten iron of the magma ocean.",
      "created_on": "2018-12-17T13:46:48.139918Z",
      "average_rating": 0,
      "updated_on": "2018-12-17T13:46:48.139938Z",
      "image_url": "money.png",
      "author": {
        "username": "naome",
        "email": "naomekizzan@gmail.com",
        "bio": "",
        "image": null,
        "following": false
      },
      "favourites_count": 0,
      "tags": [
        "Science",
        "Astronomy"
      ],
      "view_counts": 0,
      "read_time": "2 min read"
    }
  ]
}
```
#### Any background context you want to provide?
- The read time is calculated automatically.
- Each article is returned with its read time.

#### What are the relevant pivotal tracker stories?
#162163014
